src/common/pick_address.cc

```diff
@@ -164,27 +164,34 @@ void pick_addresses(CephContext *cct, int needs)
     exit(1);
   }
 
+  auto& paddr = cct->_conf->get_val<entity_addr_t>("public_addr");
+  const std::string& pnetwork =
+    cct->_conf->get_val<std::string>("public_network");
+  auto& caddr = cct->_conf->get_val<entity_addr_t>("cluster_addr");
```

Comment:
please move the definition to where it's used for the first time. and i'd suggest use descriptive names like `public_addr` instead of `paddr` if possible for better readability.